packages/firestore/src/local/local_documents_view.ts

```diff
@@ -227,24 +227,19 @@ export class LocalDocumentsView {
         for (const batch of batches) {
           batch.keys().forEach(key => {
             const baseDoc = docs.get(key);
-            if (baseDoc !== null) {
-              let mask: FieldMask | null = masks.has(key)
-                ? masks.get(key)!
-                : FieldMask.empty();
-              mask = batch.applyToLocalView(baseDoc, mask);
-              masks.set(key, mask);
-              if (documentsByBatchId.get(batch.batchId) === null) {
-                documentsByBatchId = documentsByBatchId.insert(
-                  batch.batchId,
-                  documentKeySet()
-                );
-              }
-              const newSet = documentsByBatchId.get(batch.batchId)!.add(key);
-              documentsByBatchId = documentsByBatchId.insert(
-                batch.batchId,
-                newSet
-              );
+            if (baseDoc === null) {
+              return;
             }
+            let mask: FieldMask | null = masks.get(key) ?? FieldMask.empty();
```

Comment:
FWIW I know this is my suggested code but today's Sebastian would suggest `masks.get(key) || FieldMask.empty()`. The end behavior in this case is the same, but the transpiled JS is much shorter.

`a || b` translates to just `a || b`, but unfortunately also returns 'b' for any value that evaluates to `false` (0, "", null, undefined). `a ?? b` is the better version of this as it does not return `b` for 0 and "", but it translates to `a !== null && a !== void 0 ? a : b`

See https://www.typescriptlang.org/play?#code/DYUwLgBAhg3AUKSAjeiIGMIF5oQPx4QoLgQAm2uAPlUUA

(you could certainly argue that the impact on code size here is negligible).

Reply:
Done